### PR TITLE
Pkg reorg

### PR DESCRIPTION
**Pull request by jacobweinstock**
repo layout reorganization 

design

### CODE REVIEW

File: pkg/providers/interface.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package providers
+
+// Bootstrap is the interface for creating a bootstrap vm and running cluster provisioning
+type Bootstrap interface {
+	Prepare() error
```
Comment (j-griffith) [inline pkg/providers/interface.go:5]: golint will complain about exported members without docstrings here

File: pkg/providers/interface.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	Provision() error
+	Progress() error
```
Comment (bgentry-ntap) [inline pkg/providers/interface.go:10]: I'm a bit confused on Progress as an interface method. We have the events that monitor progress already right?

File: pkg/providers/interface.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	LogFile         string `yaml:"LogFile,omitempty" json:"logfile,omitempty"`
+	Addons          Addons `yaml:"Addons,omitempty" json:"addons,omitempty"`
+	Cluster         `yaml:",inline" json:",inline" mapstructure:",squash"`
+}
+
+// Cluster specifies the details about the management cluster
+type Cluster struct {
+	ClusterName           string `yaml:"ClusterName" json:"clustername"`
+	ControlPlaneCount     int    `yaml:"ControlPlaneCount" json:"controlplanecount"`
+	ControlPlaneSize      string `yaml:"ControlPlaneSize" json:"controlplanesize"`
```
Comment (bgentry-ntap) [inline pkg/providers/interface.go]: Control plane count vs size?

File: pkg/providers/interface.go (hunk 1 above)
Comment (jacobweinstock) [reply pkg/providers/interface.go:6]: updated

File: pkg/providers/interface.go (hunk 2 above)
Comment (jacobweinstock) [reply pkg/providers/interface.go:10]: events are more like log messages. `Progress` is the method that will know when the engine is either finished or errored out.

File: pkg/providers/interface.go (hunk 3 above)
Comment (jacobweinstock) [reply pkg/providers/interface.go]: the count is the number of VMs, size is how big the VM is in terms of mem, CPU, disk. that value isn't completely thought through. I can remove it if you think we should revisit this idea.

Comment (jacobweinstock) [reply pkg/providers/interface.go]: oh, I just realized I remove this from this PR. we can discuss in the new PR coming for the config spec.

File: pkg/providers/interface.go (hunk 1 above)
Comment (jacobweinstock) [reply pkg/providers/interface.go:6]: updated with docstrings

**Comment by jacobweinstock** [comment]
> LGTM, Also from the diagram, is the expectation the engines call back into the providers during the process?

in general, I don't think they would. if they did it would need to only be to the general methods of the provider and not to the bootstrap pkg of the provider. but again, in general, I don't think they should.

**Pull request merged**